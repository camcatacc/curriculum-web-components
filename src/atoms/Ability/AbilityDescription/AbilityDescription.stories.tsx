// Modules
import React from "react";

// Storybook
import { ComponentStory, ComponentMeta } from "@storybook/react";

// Elements
import AbilityDescription from "atoms/Ability/AbilityDescription/AbilityDescription";

// Template
const Template: ComponentStory<typeof AbilityDescription> = (args) => <AbilityDescription {...args} />;

// Variants
export const Standard = Template.bind({});
Standard.args = {
	children: (
		<p>
			This ability is <b>the best</b> in its class
		</p>
	)
};

// Export
export default {
	title: "Atoms/AbilityDescription",
	component: AbilityDescription
} as ComponentMeta<typeof AbilityDescription>;
