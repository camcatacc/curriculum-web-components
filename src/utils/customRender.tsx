import { ThemeProvider } from "@material-ui/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { ReactElement } from "react";
import { getMaterialTheme } from "styles/materialUi";

export const customRender = (ui: ReactElement, renderOptions?: RenderOptions & { darkMode?: boolean }): RenderResult => {
	const Wrapper: React.FC = ({ children }) => {
		return <ThemeProvider theme={getMaterialTheme(renderOptions?.darkMode ? "dark" : "light")}>{children}</ThemeProvider>;
	};

	return render(ui, { wrapper: Wrapper, ...renderOptions });
};
