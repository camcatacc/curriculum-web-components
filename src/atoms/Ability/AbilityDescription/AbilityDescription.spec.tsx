// Modules
import React from "react";
import { customRender } from "utils/customRender";

// Accessibility
import { axe, toHaveNoViolations } from "jest-axe";

// Elements
import AbilityDescription from "atoms/Ability/AbilityDescription/AbilityDescription";

// Tests
expect.extend(toHaveNoViolations);
describe("AbilityDescription", () => {
	it("Renders as expected and has no accessibility violations", async () => {
		const { container } = customRender(<AbilityDescription>Hello, hello</AbilityDescription>);
		expect(container).toMatchSnapshot();
		expect(await axe(container)).toHaveNoViolations();
	});
});
