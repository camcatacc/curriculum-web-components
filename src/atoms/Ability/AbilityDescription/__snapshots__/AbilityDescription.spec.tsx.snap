// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbilityDescription Renders as expected and has no accessibility violations 1`] = `
<div>
  Hello, hello
</div>
`;
